src/modules/cpu/host_advanced_augmentations.hpp

```diff
@@ -2620,4 +2620,144 @@ RppStatus glitch_host_batch(T* srcPtr, RppiSize *batch_srcSize, RppiSize *batch_
     return RPP_SUCCESS;
 }
 
+
+/**************** ricap - random image crop and patch ***************/
+
+template <typename T>
+RppStatus ricap_host_batch(T* srcPtr, RppiSize *batch_srcSize, RppiSize *batch_srcSizeMax, T* dstPtr,
+                                    Rpp32u *permutedIndices1, Rpp32u *permutedIndices2, Rpp32u *permutedIndices3, Rpp32u *permutedIndices4,
+                                    Rpp32u *cropRegion1, Rpp32u *cropRegion2, Rpp32u *cropRegion3, Rpp32u *cropRegion4,
+                                    Rpp32u outputFormatToggle, Rpp32u nbatchSize,
+                                    RppiChnFormat chnFormat, Rpp32u channel)
+{
+
+    if(chnFormat == RPPI_CHN_PLANAR)
+    {
+        //to planar
+        Rpp32u cropBufferinBytes[4];
+        cropBufferinBytes[0] = cropRegion1[2] * sizeof(T);
+        cropBufferinBytes[1] = cropRegion2[2] * sizeof(T);
+        cropBufferinBytes[2] = cropRegion3[2] * sizeof(T);
+        cropBufferinBytes[3] = cropRegion4[2] * sizeof(T);
+
+        omp_set_dynamic(0);
+        #pragma omp parallel for num_threads(nbatchSize)
+        for(int batchCount = 0; batchCount < nbatchSize; batchCount ++)
+        {
+            Rpp32u srcAndDstImageDimMax = batch_srcSizeMax[batchCount].height * batch_srcSizeMax[batchCount].width;
+            T *srcPtrImage1,*srcPtrImage2,*srcPtrImage3,*srcPtrImage4, *dstPtrImage;
+            Rpp32u srcLoc1 = 0,srcLoc2 = 0,srcLoc3 = 0,srcLoc4 = 0, dstLoc = 0;
+            compute_image_location_host(batch_srcSizeMax, permutedIndices1[batchCount], &srcLoc1, channel);
+            compute_image_location_host(batch_srcSizeMax, permutedIndices2[batchCount], &srcLoc2, channel);
+            compute_image_location_host(batch_srcSizeMax, permutedIndices3[batchCount], &srcLoc3, channel);
+            compute_image_location_host(batch_srcSizeMax, permutedIndices4[batchCount], &srcLoc4, channel);
+            compute_image_location_host(batch_srcSizeMax, batchCount, &dstLoc, channel);
+            srcPtrImage1 = srcPtr + srcLoc1;
+            srcPtrImage2 = srcPtr + srcLoc2;
+            srcPtrImage3 = srcPtr + srcLoc3;
+            srcPtrImage4 = srcPtr + srcLoc4;
+            dstPtrImage = dstPtr + dstLoc;
+            Rpp32u width_srcSizeMax = batch_srcSizeMax[permutedIndices1[0]].width;
+
+            for(int c = 0; c < channel; c++)
+            {
+                T *dstPtrChannel, *srcPtrChannelROI1, *srcPtrChannelROI2, *srcPtrChannelROI3, *srcPtrChannelROI4;
+                dstPtrChannel = dstPtrImage + (c * srcAndDstImageDimMax);
```

Comment:
For the planar1/planar3 type, once you get the pointer to ROI location in channel 1, the channel 2 and channel 3 ROI locations are always going to be an increment of srcAndDstImageDimMax away from the previous channel. You don't need the computations in L2665-L2669 3 times inside the for loop.